libs/ng-strapi-auth/src/lib/core/services/user/user.service.ts

```diff
@@ -1,9 +1,39 @@
 import { Injectable } from '@angular/core';
+import { map } from 'rxjs/operators';
+import { AuthService } from '../auth/auth.service';
+import { Observable } from 'rxjs';
+import { UserModel, DefaultUserModel } from '../../models';
+import { LocalStorageKeyEnum } from '../../enums';
 
 @Injectable({
-  providedIn: 'root'
+    providedIn: 'root'
 })
 export class UserService {
 
-  constructor() { }
+    constructor(
+        public authService: AuthService,
+    ) {
+        this.authService.authUserChanged$.subscribe((user) => {
+            if (user) {
+                this.setUser(user)
+            } else {
+                this.removeUser()
+            }
+        })
+    }
+
+    setUser(user: UserModel) {
+        sessionStorage.setItem(LocalStorageKeyEnum.CURRENT_USER, JSON.stringify(user));
+    }
+
+    removeUser() {
+        sessionStorage.removeItem(LocalStorageKeyEnum.CURRENT_USER);
+    }
+
+    getCurrentUser<T = DefaultUserModel>(): Observable<UserModel<T | DefaultUserModel> | null> {
```

Comment:
```suggestion
    getCurrentUser<T = DefaultUserModel>(): Observable<UserModel<T>> {
```
That'll be enough, T is already assigned at the beginning